presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/StreamPropertyDerivations.java

```diff
@@ -0,0 +1,566 @@
+
+            throw new UnsupportedOperationException("not yet implemented");
+        }
+
+        //
+        // Nodes that rewrite and/or drop symbols
+        //
+
+        @Override
+        public StreamProperties visitProject(ProjectNode node, List<StreamProperties> inputProperties)
+        {
+            StreamProperties properties = Iterables.getOnlyElement(inputProperties);
+
+            // Only identity projected symbols are passed through
+            Map<Symbol, Symbol> identities = computeIdentityTranslations(node.getAssignments());
+            return properties.translate(column -> Optional.ofNullable(identities.get(column)));
+        }
+
+        private static Map<Symbol, Symbol> computeIdentityTranslations(Map<Symbol, Expression> assignments)
+        {
+            Map<Symbol, Symbol> inputToOutput = new HashMap<>();
+            for (Map.Entry<Symbol, Expression> assignment : assignments.entrySet()) {
+                if (assignment.getValue() instanceof QualifiedNameReference) {
+                    inputToOutput.put(Symbol.fromQualifiedName(((QualifiedNameReference) assignment.getValue()).getName()), assignment.getKey());
+                }
+            }
+            return inputToOutput;
+        }
+
+        @Override
+        public StreamProperties visitAggregation(AggregationNode node, List<StreamProperties> inputProperties)
+        {
+            StreamProperties properties = Iterables.getOnlyElement(inputProperties);
+
+            // Only grouped symbols projected symbols are passed through
+            return properties.translate(symbol -> node.getGroupBy().contains(symbol) ? Optional.of(symbol) : Optional.<Symbol>empty());
+        }
+
+        @Override
+        public StreamProperties visitTableFinish(TableFinishNode node, List<StreamProperties> inputProperties)
+        {
+            StreamProperties properties = Iterables.getOnlyElement(inputProperties);
+            // table commit does not output any symbols
+            return properties.withoutSymbols();
+        }
+
+        @Override
+        public StreamProperties visitDelete(DeleteNode node, List<StreamProperties> inputProperties)
+        {
+            StreamProperties properties = Iterables.getOnlyElement(inputProperties);
+            // delete does not output any symbols
+            return properties.withoutSymbols();
+        }
+
+        @Override
+        public StreamProperties visitTableWriter(TableWriterNode node, List<StreamProperties> inputProperties)
+        {
+            StreamProperties properties = Iterables.getOnlyElement(inputProperties);
+            // writer does not output any symbols
+            return properties.withoutSymbols();
+        }
+
+        @Override
+        public StreamProperties visitUnnest(UnnestNode node, List<StreamProperties> inputProperties)
+        {
+            StreamProperties properties = Iterables.getOnlyElement(inputProperties);
+
+            // only some symbols are passed through
```

Comment:
Similar to the comment above re: project. I'd say something about being able to describe properties only in terms of columns that are projected unmodified.